File: src/test/java/com/gemini/krakenbot/config/SettingsTest.java
package com.gemini.krakenbot.config;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class SettingsTest {

    @Test
    void constructor_defaultsNullValues() {
        Settings settings = new Settings(10L, 1.5, null, true, null, null);
        assertEquals(5.0, settings.dustThresholdUSD());
        assertEquals(0.0, settings.fiatMaxDrawdown());
        assertEquals(1.0, settings.fiatDeploymentExponent());
        assertTrue(settings.dryRun());
        assertEquals(10L, settings.loopDelaySeconds());
        assertEquals(1.5, settings.deviationTriggerPercent());
    }

    @Test
    void constructor_retainsNonNullValues() {
        Settings settings = new Settings(20L, 2.5, 10.0, false, 15.0, 2.0);
        assertEquals(10.0, settings.dustThresholdUSD());
        assertEquals(15.0, settings.fiatMaxDrawdown());
        assertEquals(2.0, settings.fiatDeploymentExponent());
        assertFalse(settings.dryRun());
        assertEquals(20L, settings.loopDelaySeconds());
        assertEquals(2.5, settings.deviationTriggerPercent());
    }
}

File: src/test/kotlin/com/gemini/krakenbot/config/SettingsTest.kt
package com.gemini.krakenbot.config

import com.gemini.krakenbot.TestFixtures
import io.kotest.core.spec.IsolationMode
import io.kotest.core.spec.style.StringSpec
import io.kotest.matchers.booleans.shouldBeTrue
import io.kotest.matchers.shouldBe

class SettingsTest : StringSpec() {
    override fun isolationMode() = IsolationMode.InstancePerTest

    init {
        "constructor_defaultsNullValues" {
            val settings =
                TestFixtures.settings(loopDelaySeconds = 10L, deviationTriggerPercent = 1.5, dustThresholdUSD = 5.0)
            settings.dustThresholdUSD shouldBe 5.0
            settings.fiatMaxDrawdown shouldBe 0.0
            settings.fiatDeploymentExponent shouldBe 1.0
            settings.dryRun.shouldBeTrue()
            settings.loopDelaySeconds shouldBe 10L
            settings.deviationTriggerPercent shouldBe 1.5
        }

        "constructor_retainsNonNullValues" {
            val settings =
                TestFixtures.settings(
                    dryRun = false,
                    loopDelaySeconds = 20L,
                    deviationTriggerPercent = 2.5,
                    dustThresholdUSD = 10.0,
                    fiatMaxDrawdown = 15.0,
                    fiatDeploymentExponent = 2.0,
                )
            settings.dustThresholdUSD shouldBe 10.0
            settings.fiatMaxDrawdown shouldBe 15.0
            settings.fiatDeploymentExponent shouldBe 2.0
            settings.dryRun shouldBe false
            settings.loopDelaySeconds shouldBe 20L
            settings.deviationTriggerPercent shouldBe 2.5
        }
    }
}
